### DISCUSSION TITLE
Add support for SVG images

### DISCUSSION THREAD

**Discussion by caygri** [PrestaShop/PrestaShop#32240] [Ideas]
**Is your feature request related to a problem? Please describe.**
Hello, i have a lot of requesto about integrate svg logo and libraries, especially for design product

**Describe alternatives you've considered**
i cant considered because need preview of image.

---

Edited on November 24th, following https://github.com/PrestaShop/PrestaShop/issues/10600#issuecomment-728133716:

**Describe the solution you'd like:**

1. Add SVG as a supported image format.
2. Define its MIME type in ImageManager.
3. Allow the upload of SVG images.

**Specifications:**

https://github.com/PrestaShop/prestashop-specs/pull/179

**Comment by khouloudbelguith** [comment]
Hi @caygri,

Your ticket is a duplicate of #10290
So, I close it.
Thanks!

Best regards, Khouloud

**Comment by khouloudbelguith** [comment]
Duplicate of #10290

**Comment by Klemart3D** [comment]
I don't think it's a duplicate.
 #10290 ticket talking about allowing .png format for favicon upload (only).
This ticket talking about allow SVG format file globally in PrestaShop, specially for picture file or thumbnail upload for a product (when we've got Symfony error: "product.svg: This file is not a valid image" in PS v1.7) or for categories or modules like _ps_banner_, _ps_imageslider_ or _blockreassurance_ ("Image format not recognized, allowed formats are: .gif, .jpg, .png").
I'm also creating a module to upload custom pictograms with the HelperForm "type => file", but SVG is still unfortunately unsupported because PrestaShop uses PHP's [GD library](https://secure.php.net/manual/en/intro.image.php).

![error-svg](https://user-images.githubusercontent.com/7419525/54593808-9d085c80-4a2f-11e9-8b0b-80d5d2b9d34a.png)

**Comment by dennispw** [comment]
This would be a great addition.

**Comment by khouloudbelguith** [comment]
Hi @Klemart3D,

Thanks for your feedback.
Yes, I tried to import an SVG image with PS1.7.5.1, but I have an issue.
![image](https://user-images.githubusercontent.com/16067358/54601312-05ab0580-4a3f-11e9-82c2-02e55b3ded6e.png)
@marionf, @colinegin what do you think? it could an improvement issue to accept SVG image.
Related to #10290

Thanks!

**Comment by caygri** [comment]
it's impossible because prestashop use an old framwork who not supported

**Comment by MatShir** [comment]
@PrestaShop/prestashop-core-developers is it really impossible?
SVG could enhance the quality of some merchant website but it needs to be used carefully.

**Comment by PierreRambaud** [comment]
Yes, You must use Imagick instead of GD.
But there should be some alternatives.

**Comment by jolelievre** [comment]
Maybe we could add support for imagick in PrestaShop, which would open new allowed formats

**Comment by wdammak** [comment]
hh svg not supported??? why!! :'(
svg == the future of the web for images

**Comment by caygri** [comment]
because they use smarty and smarty non support svg

**Comment by wdammak** [comment]
@caygri no i don't think it's smarty related!
[https://stackoverflow.com/questions/54950670/prestashop-1-7-include-inline-svg-in-tpl-file](https://stackoverflow.com/questions/54950670/prestashop-1-7-include-inline-svg-in-tpl-file)

**Comment by PierreRambaud** [comment]
It's because of the legacy implementation, many things need to be refactor to allow the usage of new image type.

**Comment by wdammak** [comment]
it's been almost 10 years that the svg is supported by the majority of browsers, I don't understand why the team has found it difficult to integrate a new mime type "image/svg+xml" during an image upload!
A svg file is considered exactly as a png or jpg file in its use and display!

**Comment by matks** [comment]
> it's been almost 10 years that the svg is supported by the majority of browsers, I don't understand why the team has found it difficult to integrate a new mime type "image/svg+xml" during an image upload!

Because this is only one of the items required to allow full SVG implementation in prestashop. Images in PrestaShop are moved, copied, resized, displayed in many manners ; and the functions required to perform all of these processings are currently being performed by PHP GD which does not support SVG.

So the work needed is a lot more, and currently are hands are full with other topics (see the in-progress work here: https://github.com/PrestaShop/PrestaShop/projects/7) but if you are motivated, would you give us a hand ? this is an opensource project, everybody can contribute 😄

**Comment by caygri** [comment]
@matks  I will like to help, but the problem is the core of prestashop becuase smarty not support svg. 
One year ago I will talk with Aurelien and he said is not possible and not see the urgent to change. 

I have a client and i need migrate to magento for this missing part!

**Comment by tameroski** [comment]
Smarty is a templating engine, it has nothing to do with SVG.

**Comment by wdammak** [comment]
> Because this is only one of the items required to allow full SVG implementation in prestashop. Images in PrestaShop are moved, copied, resized, displayed in many manners ; and the functions required to perform all of these processings are currently being performed by PHP GD which does not support SVG.
> 
> So the work needed is a lot more, and currently are hands are full with other topics (see the in-progress work here: https://github.com/PrestaShop/PrestaShop/projects/7) but if you are motivated, would you give us a hand ? this is an opensource project, everybody can contribute smile

Exactly the advantage is that you no longer need to create/copy/place in a specific folder ... as prestashop currently does. Just use scaleImage to have the desired size and the file size is always the same, it's also the same to change the color according to the theme...

`$image = new Imagick();
$image->setBackgroundColor(new ImagickPixel('green'));
$image->readImage('./some/path/image.svg');
$image->scaleImage(20,20);`

**Comment by jf-viguier** [comment]
Svg are vectorial, there is no need for thumbnail anymore : no copy or resize, Imagick not needed.

**Comment by jf-viguier** [comment]
By the way, if you really wants to change size of a svg for thumbailing, it's just an xml file. This can be done with few lines of php, no imagick : https://stackoverflow.com/a/48831185

**Comment by LouiseBonnard** [comment]
@Progi1984 and I discussed this morning the possibility to add the SVG image format to PrestaShop. We decided to focus on all images upload available in the back office, except for the favicon and the email logos in the Design > Theme & Logo page.

Why? First, because favicons are a specific case using precise image formats and then because the email support for SVG images is way too limited for now.

<img width="1342" alt="Capture d’écran 2020-11-16 à 16 25 09" src="https://user-images.githubusercontent.com/32565890/99272704-65673e80-2828-11eb-98da-7849a7fa2698.png">

So here are the identified steps to offer SVG support in PrestaShop:

1. Add **SVG as a supported** image format.
2. **Define its MIME type** in ImageManager.
3. **Allow the upload** of SVG images.

**Comment by caygri** [comment]
so now is avaible?

**Comment by LouiseBonnard** [comment]
> so now is available?

I hope for the 1.7.8 version! Feel free to let us know if you'd like to create a pull request to make it available. ;-)

**Comment by LouiseBonnard** [comment]
As discussed with @PierreRambaud yesterday, PrestaShop should not allow the SVG format by default. It should rather be enabled by the merchant instead. It implies adding an option to the _Images Settings_ page of the back office to let the users activate it if they want to. @Julievrz, any idea for the wording? Many thanks in advance!

**Comment by Julievrz** [comment]
Here is my suggestion: 
'Enable SVG' in Admin.Design.Feature (No by default)
'Note that this image format is not supported for the favicon and the email logo in the Design > Theme & Logo page'. in Admin.Design.Help

I think it could be located just under "Image format".

**Comment by MatShir** [comment]
On the Block reassurance check upload SVG !

**Comment by alfonsolr09** [comment]
what happens with svg?

**Comment by laurentrousseau918** [comment]
+1

**Comment by MatShir** [comment]
With this new feature coming -> https://github.com/PrestaShop/PrestaShop/pull/30045, we can add multiple format at the same time and configure a fallback if the browser doesn't support the image format. 
The PR includes the support of AVIF, but for the future I will be easier to add new format. Don't hesitate to review the work

**Comment by cruftex** [comment]
@eternoendless Hi Pablo,

The original issue is part of the epic https://github.com/PrestaShop/PrestaShop/issues/18518. In the Epic "Add support for SVG image" now shows as completed. It would be good to not close the issue as completed when transferring.

However, why was this moved to discussion? Does it mean it is not a valid issue / feature request?

**Reply by MatShir** [reply]
Good catch! I'm updating the link with the link of the discussions

**Comment by cruftex** [comment]
Current status of relevant SVG support:

### Web / browsers

okay except favicon, https://caniuse.com/?search=svg

### PDF

the used PDF engine, TCPDF, supports SVG, with PS 8.0.4 I managed to embed a SVG in the invoice by adding it to the invoice template directly

### email

Gmail does not support SVG properly, https://www.caniemail.com/features/image-svg/, and for embedding: https://www.caniemail.com/features/html-svg/. For the moment it seems the best option is to use a pixel image as fallback.

**Comment by ludini17** [comment]
+1
